presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/IOPlanPrinter.java

```diff
@@ -522,7 +522,10 @@ else if (writerTarget instanceof CreateName || writerTarget instanceof InsertRef
             ImmutableSet.Builder<ColumnConstraint> columnConstraints = ImmutableSet.builder();
             for (Map.Entry<ColumnHandle, Domain> entry : constraint.getDomains().get().entrySet()) {
                 ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, entry.getKey());
-                columnConstraints.add(new ColumnConstraint(columnMetadata.getName(), columnMetadata.getType().getTypeSignature(), parseDomain(entry.getValue())));
+                columnConstraints.add(new ColumnConstraint(
+                        columnMetadata.getName(),
+                        columnMetadata.getType().getTypeSignature(),
+                        parseDomain(entry.getValue().simplify())));
```

Comment:
Consider adding a test and some explanation to the commit message.

Reply:
Done. Thanks for the suggestion!